plugins/Microwave/Microwave.cpp

```diff
@@ -0,0 +1,6143 @@
+				for (int i = 0; i < m_oversample.value() + 1; ++i)
+				{
+					ps->nextStringSample(outputSample, m_waveforms, m_subs, m_samples, m_sampGraphs, m_maxMainEnabled, m_maxSubEnabled, m_maxSampleEnabled, m_maxFiltEnabled, m_maxModEnabled, Engine::mixer()->processingSampleRate() * (m_oversample.value() + 1), m_mwc, m_removeDC.value(),  m_storedsubs);
+				}
+				totalOutputSample[0] = outputSample[0];
+				totalOutputSample[1] = outputSample[1];
+
+				break;
+			}
+			case 1:
+			{
+				// Process some number of samples and average them together, depending on the oversampling value.
+				// For example, if the oversampling is set to 4x, it will calculate 4 samples and take the average of those.
+				for (int i = 0; i < m_oversample.value() + 1; ++i)
+				{
+					ps->nextStringSample(outputSample, m_waveforms, m_subs, m_samples, m_sampGraphs, m_maxMainEnabled, m_maxSubEnabled, m_maxSampleEnabled, m_maxFiltEnabled, m_maxModEnabled, Engine::mixer()->processingSampleRate() * (m_oversample.value() + 1), m_mwc, m_removeDC.value(), m_storedsubs);
+					totalOutputSample[0] += outputSample[0];
+					totalOutputSample[1] += outputSample[1];
+				}
+
+				totalOutputSample[0] /= (m_oversample.value() + 1);
+				totalOutputSample[1] /= (m_oversample.value() + 1);
+
+				break;
+			}
+		}
+
+		for (ch_cnt_t chnl = 0; chnl < DEFAULT_CHANNELS; ++chnl)
+		{
+			//Send to output
+			working_buffer[frame][chnl] = totalOutputSample[chnl];
+		}
+
+		// Update visualizer
+		if (m_viewOpen && m_visualize.value() && m_scroll == 0 && ps->m_enabled[m_mainNum.value()-1])
+		{
+			m_visualizerValues[int((ps->m_sample_realindex[m_mainNum.value()-1][0] / (ps->m_sampLen[m_mainNum.value()-1] * WAVERATIO)) * 204.f)] = ps->m_mainsample[m_mainNum.value()-1][0] * m_visvol.value() * 0.01f;
+			if (ps->m_noteDuration % 1470 == 1)// Updates around 30 times per second per note
+			{
+				m_graph.setSamples(m_visualizerValues);
+			}
+		}
+	}
+
+	applyRelease(working_buffer, n);
+
+	instrumentTrack()->processAudioBuffer(working_buffer, frames + offset, n);
+}
+
+
+void Microwave::deleteNotePluginData(NotePlayHandle * n)
+{
+}
+
+
+// Fill m_subs using m_storedsubs, oftentimes making use of libsamplerate for some awesome sinc interpolation.
+inline void Microwave::fillSubOsc(int which, bool doInterpolate)
+{
+	if (doInterpolate)
+	{
+		srccpy(m_subs[which].data(), m_storedsubs[which].data(), STOREDSUBWAVELEN);
+	}
+	else
+	{
+		for (int i = 0; i < STOREDSUBWAVELEN; ++i)
+		{
+			for (int j = 0; j < WAVERATIO; ++j)
+			{
+				m_subs[which][i*WAVERATIO+j] = m_storedsubs[which][i];
+			}
+		}
+	}
+	m_subFilled[which] = true;
+}
+
+
+
+// Fill m_waveforms using m_storedwaveforms, oftentimes making use of libsamplerate for some awesome sinc interpolation.
+inline void Microwave::fillMainOsc(int which, bool doInterpolate)
+{
+	if (doInterpolate)
+	{
+		srccpy(m_waveforms[which].data(), m_storedwaveforms[which].data(), STOREDMAINARRAYLEN);
```

Comment:
I explained that because it may work in ways that people don't think of as a slope increase (e.g. Allpass filter with a higher slope), but I do agree that only the first half of that tooltip is necessary.